File: Frontend/src/app/Models/Pago.ts
import { CuentaHabitacion } from './CuentaHabitacion';
import { MetodoPago } from './MetodoPago';
import { Reserva } from './Reserva';

export interface Pago {
  idPago: number;
  monto: number;
  fecha: string | Date;
  estado: string;
  reserva: Reserva;
  metodoPago: MetodoPago;
}

File: Backend/src/main/java/com/example/lumaresort/entities/Pago.java
package com.example.lumaresort.entities;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Pago {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idPago;

    private float monto;
    private Date fecha;
    private String estado;

    /*@JsonIgnore
    @ManyToOne
    private CuentaHabitacion cuentaHabitacion;
     */
    private String metodoPago;

    @OneToOne
    private Reserva reserva;
}
